### DISCUSSION TITLE
Disable cache? (tsml-cache-f39852333f.json)

### DISCUSSION THREAD

**Discussion by richieboo** [code4recovery/12-step-meeting-list#1315] [Q&A] [answered]
Anyone know if there is a way to disable the cache json file that is in the wp-content folder? I keep having to delete it so that deleted meetings don't continue to show in the search results. 

/public_html/wp-content/tsml-cache-f39852333f.json

**Comment by tim-rohrer** [comment]
Hey @richieboo,

We believe something odd is going on with your site. TSML will rewrite the cache file every time a meeting is updated, and that includes being deleted. This rewrite process works on hundreds of sites, and we've not had reports of this issue any time in the recent past.

I suspect something on your site is interfering with the rewrite process. It could be as simple as permissions. 

When you delete a meeting (probably ought to test with a simple edit as well), watch the developer/debug console for any errors. If you don't see any, study the permissions to see if something got set incorrectly at some point.

Hopefully that will get you started in troubleshooting.

**Reply by tim-rohrer** [reply]
I guess I'm also curious to hear more about what search tool you're using that is showing meetings that you deleted. And wondering if perhaps some caching plugin you've got installed is creating the issue? Maybe focus more of that than the steps I suggested above.

**Reply by richieboo** [reply]
I'm using the search within the TSML plugin. It's probably permissions so I'll check that and report back here what I find.

Currently, I edit the meeting, choose delete permanently, then go back to meetings page and search for the meeting. It shows up. If I try to edit the meeting again, it alerts me I can't edit a deleted meeting. Then I delete the cache file, search for meeting, and it's not found.

**Reply by tim-rohrer** [reply]
Were there any error messages in the console?

**Reply by joshreisner** [reply]
if you add a new meeting, or edit one, do you also need to remove the cache file manually to see the change on the frontend?

**Comment by richieboo** [comment]
No, I see the edit there right away.

My steps I just tried:
Edit meeting, add a period to meeting post title
Save meeting
Search meetings again and in results grid the period shows immediately, as well as viewing the individual meeting.

**Reply by joshreisner** [reply]
ok interesting, good data point

> edit the meeting, choose delete permanently

when i delete a meeting, i "move to trash" - is that what you do?

<img width="409" alt="Screenshot 2024-01-19 at 2 26 13 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/005c78cd-e929-4b63-b109-9b96c3da50d9">

**Comment by richieboo** [comment]
No, this is what I have. This HAS to be the problem..

![image](https://github.com/code4recovery/12-step-meeting-list/assets/11405496/bf2eae6c-0301-4731-b2df-c3c660e82f00)

**Comment by richieboo** [comment]
I'm going to go back and remove    define('EMPTY_TRASH_DAYS', 0)   in my config. I was trying to keep meetings in the trash so I could recover them if AAs wanted to revive them. Will report back.

**Comment by richieboo** [comment, marked as answer]
Mystery solved. Some things never change, I am the problem.  (I am an alcoholic, and Richie is my problem)

I created my problem with define('EMPTY_TRASH_DAYS', 0) thinking that 0 would disable the option. Never picked up that the changing of the language to "Delete Permanently" meant what it reads.

So, just updated to define('EMPTY_TRASH_DAYS', 999999);  

But, always good to find bugs. If someone really does want to delete permanently, apparently this breaks the updating TSML cache file.

Thanks for reading!
